galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.vtp.manager/src/main/java/dev/galasa/vtp/manager/internal/VtpManagerImpl.java

```diff
@@ -0,0 +1,300 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.StringTokenizer;
+
+import javax.validation.constraints.NotNull;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.annotations.Component;
+
+import dev.galasa.ManagerException;
+import dev.galasa.Test;
+import dev.galasa.artifact.IArtifactManager;
+import dev.galasa.artifact.TestBundleResourceException;
+import dev.galasa.cicsts.CicsRegion;
+import dev.galasa.cicsts.CicstsManagerField;
+import dev.galasa.cicsts.ICicsRegion;
+import dev.galasa.cicsts.ICicsTerminal;
+import dev.galasa.cicsts.spi.ICicstsManagerSpi;
+import dev.galasa.framework.spi.AbstractManager;
+import dev.galasa.framework.spi.AnnotatedField;
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.framework.spi.IConfigurationPropertyStoreService;
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IManager;
+import dev.galasa.framework.spi.ResourceUnavailableException;
+import dev.galasa.framework.spi.language.GalasaMethod;
+import dev.galasa.framework.spi.language.GalasaTest;
+import dev.galasa.vtp.internal.properties.DataSetHLQ;
+import dev.galasa.vtp.internal.properties.TransactionNamesForTag;
+import dev.galasa.vtp.internal.properties.VtpAPI;
+import dev.galasa.vtp.internal.properties.VtpEnable;
+import dev.galasa.vtp.internal.properties.VtpPropertiesSingleton;
+import dev.galasa.vtp.manager.VtpManagerException;
+import dev.galasa.zos.IZosImage;
+import dev.galasa.zos3270.spi.IZos3270ManagerSpi;
+import dev.galasa.zos3270.spi.NetworkException;
+import dev.galasa.zosbatch.IZosBatchJob;
+import dev.galasa.zosbatch.ZosBatchException;
+import dev.galasa.zosbatch.spi.IZosBatchSpi;
+
+@Component(service = { IManager.class })
+public class VtpManagerImpl extends AbstractManager {
+
+	private static final Log logger = LogFactory.getLog(VtpManagerImpl.class);
+
+	public final static String NAMESPACE = "vtp";
+	private static final String PASSED_RESULT = "Passed";
+
+	private ICicstsManagerSpi cicsManager;
+	private IZosBatchSpi      batchManager;
+	private IArtifactManager  artifactManager;
+	private IConfigurationPropertyStoreService cps;
+	
+	private HashMap<ICicsRegion,RecordingData> recordingRegions = new HashMap<>(); 
+	
+	private String dumpDataSetHLQ  = new String();
+	private boolean skipRecordings = false;
+	
+	private Path storedArtifactRoot;
+	
+	private VtpRecorderImpl recorder;
+
+	@Override
+	public void provisionGenerate() throws ManagerException, ResourceUnavailableException {
+		List<AnnotatedField> foundAnnotatedFields = findAnnotatedFields(CicstsManagerField.class);
+		for (AnnotatedField annotatedField : foundAnnotatedFields) {
+			Field field = annotatedField.getField();
+			if (field.getType() == ICicsRegion.class) {
```

Comment:
no way.
The VTP Manager MUST not touch the cics manager annotation or fields.
If the VTP Manager requires a list of CICS regions,  then it should as for them via the SPI.

Reply:
if you need an annotaion to indicate a link to a cics region,   then the vtp manager should have it's own annotation